CLI.md

```diff
@@ -0,0 +1,105 @@
+and will accept any valid rule in the following formats:
+* toml
+* json
+* yaml (yup)
+* ndjson (as long as it contains only a single rule and has the extension `.ndjson` or `.jsonl`)
+
+#### `import-rules`
+
+The primary advantage of using this command is the ability to import multiple rules at once. Multiple rule paths can be
+specified explicitly with unlimited arguments, recursively within a directory using `-d/--directory`[*](#i18n-note-3), or 
+a combination of both.
+
+In addition to the formats mentioned using `create-rule`, this will also accept an `.ndjson`/`jsonl` file 
+containing multiple rules (as would be the case with a bulk export).
+
+This will also strip additional fields and prompt for missing required fields.
+
+<a id="i18n-note-3">\* Note</a>: This will attempt to parse ALL files recursively within a specified directory.
+
+
+## Uploading rules to Kibana
+
+Toml formatted rule files can be uploaded as custom rules using the `kibana-upload` command. To upload more than one 
+file, specify multiple files at a time as individual args.
+
+```console
+python -m detection_rules kibana-upload my-rules/example_custom_rule.toml
+```
+
+_*Note that in order to load a custom rule, the proper index must be setup first. The simplest way to do this is to click 
+the `Load prebuilt detection rules and timeline templates` button on the `detections` page in the Kibana security app._
+
+
+## Converting between JSON and TOML
+
+[Importing rules](#importing-rules-into-the-repo) will convert from any supported format to toml. Additionally, the 
+command `view-rule` will also allow you to view a converted rule without importing it by specifying the `--as-rule` flag.
+
+To view a rule in JSON format, you can also use the `view-rule` command with the `--as-api` flag, which is the default.
+(See the [note](#a-note-on-versioning-and-the-generated-json-formatted-rules) on the JSON formatted rules)
+
+Multiple rules can be converted into their JSON representation by using the `build-release` command. This will assemble 
+rules which meet the specified filters in the [packages.yml](etc/packages.yml) file and stage them in the generated 
+`releases/{release}/rules`
+directory.
+
+## A note on versioning and the generated JSON formatted rules
+
+The rule toml files exist slightly different than they do in their final state as a json file in Kibana. The files are
+white space stripped, normalized, sorted, and indented, prior to their json conversion. Everything within the `metadata`
+table is also stripped out, as this is meant for rule development and management only.
+
+Additionally, the version of the rule is added to the file prior to exporting it. This is done to restrict version bumps
+to occur intentionally right before we create a release. Version bumps are auto-incremented based on detected changes in 
+rules. This is based on the hash of the rule in the following format:
+* sorted json
+* serialized
+* b64 encoded
```

Comment:
maybe too late now, but why the encoding?

Reply:
easier to compare and troubleshoot